src/shared_modules/content_manager/src/components/executionContext.hpp

```diff
@@ -29,6 +35,46 @@ const std::string GENERIC_OUTPUT_FOLDER_PATH {std::filesystem::temp_directory_pa
 class ExecutionContext final : public AbstractHandler<std::shared_ptr<UpdaterBaseContext>>
 {
 private:
+    /**
+     * @brief Reads and returns the last offset from the database.
+     *
+     * @param context Updater context configured with the database driver.
+     * @return unsigned int Last offset from the database.
+     */
+    unsigned int getDatabaseOffset(const UpdaterBaseContext& context) const
+    {
+        unsigned int databaseOffset;
+        try
+        {
+            databaseOffset = std::stoi(context.spRocksDB->getLastKeyValue().second.ToString());
+        }
+        catch ([[maybe_unused]] const std::runtime_error& e)
+        {
+            // First execution. Set offset to zero.
+            databaseOffset = 0;
+            context.spRocksDB->put(Utils::getCompactTimestamp(std::time(nullptr)), "0");
+        }
+
+        return databaseOffset;
+    }
+
+    /**
+     * @brief Parses the offset from the input configuration.
+     *
+     * @param inputConfig Reference to the input config.
+     * @return unsigned int Non-negative offset from the input config.
+     */
+    unsigned int getConfigOffset(const nlohmann::json& inputConfig) const
+    {
+        const auto configOffset {inputConfig.at("offset").get<int>()};
+        if (configOffset < 0)
+        {
+            throw std::runtime_error {"Offset is not a non-negative number: " + std::to_string(configOffset)};
```

Comment:
Improve: The message is a bit misleading, what about?
```suggestion
            throw std::runtime_error {"Offset must be a positive number. Offset: " + std::to_string(configOffset)};
```

Reply:
I've changed the message to 

```
Offset should be a non-negative number: <offset>
```

I prefer to have just one sentence.
Also, the offset can be zero, so the word `positive` is not correct IMO.

Let me know your opinion.

Reply:
You are right, zero is in fact positive

I think the new message is okay